src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java

```diff
@@ -203,6 +204,18 @@ private void createOrUpdateShadowUser(UUID userId, User shadowUser, UserTenant u
     }
   }
 
+  private void createOrUpdateShadowUserWithSystemUserContext(UUID userId, User shadowUser, UserTenant userTenantDto) {
+    try (var context = new FolioExecutionContextSetter(contextHelper.getSystemUserFolioExecutionContext(userTenantDto.getTenantId()))) {
+      log.info("Going to create or update shadow user with id: {} in the desired tenant: {}", userId.toString(), userTenantDto.getTenantId());
+      User user = userService.getById(userId);
+      if (Objects.nonNull(user.getActive())) {
+        activateUser(user);
+      } else {
+        createActiveUserWithPermissions(shadowUser);
+      }
+    }
```

Comment:
code duplication here , we can extract this code in new method:
```suggestion
    log.info("Going to create or update shadow user with id: {} in the desired tenant: {}", userId.toString(), userTenantDto.getTenantId());
      User user = userService.getById(userId);
      if (Objects.nonNull(user.getActive())) {
        activateUser(user);
      } else {
        createActiveUserWithPermissions(shadowUser);
      }
```

Reply:
Fixed